### ISSUE TITLE
Validation set

### ISSUE THREAD

**Issue by Capuz93** [jwyang/faster-rcnn.pytorch#119]
Hi,
I'm using my own dataset for 3D object detection (peds and vehicles), and I created trainval.txt and test.txt to obtain images and annotations; everything works but I have a question: what about validation set? How could I create and use it in this net? 

However I see you use precision, recall and average precision in test phase; so in files (.pkl) created after test phase there are the values of recall and precision. Are they the values to plot the precision/recall graph?

Thanks

**Comment by jwyang** [comment]
@Capuz93 Hi, for now, our code does not explicitly separate the dataset into training and validation. You might want to instantiate another data loader to load your validation data. Change your dataloader for the customized dataset and factory.py a bit and then use another imdb_name in trainval_net.py to specify the set for validation.

**Comment by Capuz93** [comment]
Yes, I've just done it.
I have only one question: for validation phase I run the same operation of training phase except:

> fasterRCNN.eval()
> fasterRCNN.zero_grad()
> optimizer.zero_grad()
> loss.backward()
> optimizer.step()

and I left clip_gradient function.

Is it the right way?

**Comment by jwyang** [comment]
@Capuz93 you do not need to perform loss.backward() and optimizer.step()
